From 7abc94204c2dd72fd6a731a0240123e344dd2ac9 Mon Sep 17 00:00:00 2001
From: Simon Fels <simon.fels@canonical.com>
Date: Mon, 19 Oct 2015 22:13:02 +0200
Subject: [PATCH 3/4] tools: mpris-proxy: allow user to specify different mpris
 service name

In some environments the MPRIS service isn't available on the bus name
org.mpris and for those situations we should allow the user to use
a different bus name on startup.
---
 tools/mpris-proxy.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/tools/mpris-proxy.c b/tools/mpris-proxy.c
index bf8148f..613ffa7 100644
--- a/tools/mpris-proxy.c
+++ b/tools/mpris-proxy.c
@@ -65,6 +65,7 @@ static GSList *transports = NULL;
 
 static gboolean option_version = FALSE;
 static gboolean option_export = FALSE;
+static char *option_service = NULL;
 
 struct tracklist {
 	GDBusProxy *proxy;
@@ -595,7 +596,7 @@ static gboolean name_owner_changed(DBusConnection *conn,
 		return DBUS_HANDLER_RESULT_NOT_YET_HANDLED;
 	}
 
-	if (!g_str_has_prefix(name, "org.mpris"))
+	if (!g_str_has_prefix(name, option_service))
 		return DBUS_HANDLER_RESULT_NOT_YET_HANDLED;
 
 	if (*new == '\0') {
@@ -677,7 +678,7 @@ static void parse_list_names(DBusConnection *conn, DBusMessageIter *args)
 
 		dbus_message_iter_get_basic(&array, &name);
 
-		if (!g_str_has_prefix(name, "org.mpris"))
+		if (!g_str_has_prefix(name, option_service))
 			goto next;
 
 		owner = get_name_owner(conn, name);
@@ -740,6 +741,8 @@ static void usage(void)
 }
 
 static GOptionEntry options[] = {
+	{ "service", 's', 0, G_OPTION_ARG_STRING, &option_service,
+				"Specifiy the mpris service name. Defaults to 'org.mpris'" },
 	{ "version", 'v', 0, G_OPTION_ARG_NONE, &option_version,
 				"Show version information and exit" },
 	{ "export", 'e', 0, G_OPTION_ARG_NONE, &option_export,
@@ -2533,6 +2536,9 @@ int main(int argc, char *argv[])
 		exit(0);
 	}
 
+	if (!option_service)
+		option_service = "org.mpris";
+
 	main_loop = g_main_loop_new(NULL, FALSE);
 
 	sys = g_dbus_setup_bus(DBUS_BUS_SYSTEM, NULL, NULL);
-- 
2.5.0

